include/envoy/thread/thread.h

```diff
@@ -10,6 +10,17 @@
 namespace Envoy {
 namespace Thread {
 
+class ThreadId {
+public:
+  virtual ~ThreadId() {}
+
+  virtual std::string string() const PURE;
```

Comment:
This gets used by the GuardDog to log an error in PANIC: https://github.com/envoyproxy/envoy/blob/43fc77901ee8a1608239daaa51b531f922628083/source/server/guarddog_impl.cc#L68

Reply:
I see...so the reason to call it 'string()', fmt::format will call it automatically without having to write out
```         PANIC(fmt::format("GuardDog: one thread ({}) stuck for more than watchdog_kill_timeout",
                          watched_dog.dog_->threadId()->debugString());
```
is that right? Do we do that elsewhere in Envoy? My personal preference is to have the explicit call but I'm OK with doing it this way if you put in a comment that's why it has to be called string().

Also I'm wondering if that works as you expect because threadId() actually returns a unique_ptr.

Reply:
Oh I see...sorry I had confused a few different blocks of code, and I did comment on this above as well. WatchDog::threadId() returns a std::string and I think it'd be better to return a ThreadId and convert it if needed in the panic call.